azext_iot/tests/iothub/message_endpoint/test_iothub_message_endpoint_unit.py

```diff
@@ -0,0 +1,938 @@
+# coding=utf-8
+# --------------------------------------------------------------------------------------------
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License. See License.txt in the project root for license information.
+# --------------------------------------------------------------------------------------------
+
+import pytest
+import azext_iot.iothub.commands_message_endpoint as subject
+from azure.cli.core.azclierror import (
+    MutuallyExclusiveArgumentError,
+    ResourceNotFoundError
+)
+from azext_iot.iothub.common import BYTES_PER_MEGABYTE, AuthenticationType
+from azext_iot.tests.generators import generate_names
+
+from azure.mgmt.iothub.models import ManagedIdentity
+
+hub_name = "hubname"
+hub_rg = "hubrg"
+endpoint_name = generate_names()
+resource_not_found_error = "Resource not found."
+generic_response = {generate_names(): generate_names()}
+
+
+path_find_resource = "azext_iot.iothub.providers.discovery.IotHubDiscovery.find_resource"
+arg_check_path = "azext_iot.iothub.providers.message_endpoint.MessageEndpoint._connection_string_retrieval_args_check"
+get_eventhub_cstring_path = "azext_iot.iothub.providers.message_endpoint.get_eventhub_cstring"
+get_servicebus_topic_cstring_path = "azext_iot.iothub.providers.message_endpoint.get_servicebus_topic_cstring"
+get_servicebus_queue_cstring_path = "azext_iot.iothub.providers.message_endpoint.get_servicebus_queue_cstring"
+get_cosmos_db_cstring_path = "azext_iot.iothub.providers.message_endpoint.get_cosmos_db_cstring"
+parse_cosmos_db_cstring_path = "azext_iot.iothub.providers.message_endpoint.parse_cosmos_db_connection_string"
+get_storage_cstring_path = "azext_iot.iothub.providers.message_endpoint.get_storage_cstring"
+
+
+@pytest.fixture()
+def fixture_update_endpoint_ops(mocker):
+    # Fetch connection string
+    arg_check = mocker.patch(arg_check_path)
```

Comment:
Would really like feedback here:
1. is it fine that the requests aren't made (as we aren't matching to a mocked response; if we did so, we would need to mock the entire hub object, not just a small part like here)
2. should we have each test mock its own endpoint list? Or should we keep it as it is now (creating all endpoint types at once and using that for tests)

Reply:
also regarding 1, by doing this we are bypassing the need to check if the sdk supports cosmos db or not (since we catch the call right before the sdk is called)